File: app/src/main/java/com/experiments/dineatmytime/network/ServerResponse.java
package com.experiments.dineatmytime.network;

import com.experiments.dineatmytime.model.Restaurant;
import com.experiments.dineatmytime.model.User;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class ServerResponse {


    @SerializedName("error")
    @Expose
    private Boolean error;

    @SerializedName("message")
    @Expose
    private String message;

    @SerializedName("user")
    @Expose
    private User user;

    @SerializedName("restaurants")
    @Expose
    private List<Restaurant> restaurantList;

    public Boolean getError() {
        return error;
    }

    public void setError(Boolean error) {
        this.error = error;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }


    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }


    public List<Restaurant> getRestaurantList() {
        return restaurantList;
    }

    public void setRestaurantList(List<Restaurant> restaurantList) {
        this.restaurantList = restaurantList;
    }
}

File: app/src/main/java/com/experiments/dineatmytime/network/ServerResponse.kt
package com.experiments.dineatmytime.network

import com.experiments.dineatmytime.model.*
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class ServerResponse {
    @SerializedName("error")
    @Expose
    var error: Boolean? = null

    @SerializedName("message")
    @Expose
    var message: String? = null

    @SerializedName("user")
    @Expose
    var user: User? = null

    @SerializedName("restaurants")
    @Expose
    var restaurantList: List<Restaurant>? = null

    @SerializedName("restaurant")
    @Expose
    var restaurantDetails: RestaurantDetails? = null

    @SerializedName("menus")
    @Expose
    var menuList: List<Menu>? = null

    @SerializedName("tables")
    @Expose
    var tableList: List<Table>? = null

    @SerializedName("bookings")
    @Expose
    var bookingList: List<Booking>? = null

    @SerializedName("offers")
    @Expose
    var offerList: List<Offer>? = null
}